src/ConnectedMode/Binding/ImportBeforeInstallTrigger.cs

```diff
@@ -29,21 +31,38 @@ namespace SonarLint.VisualStudio.ConnectedMode.Binding
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class ImportBeforeInstallTrigger
     {
-        private readonly IConfigurationProvider configurationProvider;
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
         private readonly IImportBeforeFileGenerator importBeforeFileGenerator;
         private readonly IThreadHandling threadHandling;
 
+        private bool disposed;
+
         [ImportingConstructor]
-        public ImportBeforeInstallTrigger(IConfigurationProvider configurationProvider, IImportBeforeFileGenerator importBeforeFileGenerator, IThreadHandling threadHandling)
+        public ImportBeforeInstallTrigger(IActiveSolutionBoundTracker activeSolutionBoundTracker, IImportBeforeFileGenerator importBeforeFileGenerator, IThreadHandling threadHandling)
         {
-            this.configurationProvider = configurationProvider;
+            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
             this.importBeforeFileGenerator = importBeforeFileGenerator;
             this.threadHandling = threadHandling;
+
+            this.activeSolutionBoundTracker.PreSolutionBindingChanged += OnPreSolutionBindingChanged;
+            this.activeSolutionBoundTracker.PreSolutionBindingUpdated += OnPreSolutionBindingUpdated;
+
+            // Trigger an initial update of importBeforeFileGenerator (we might have missed the solution binding
+            // event from the ActiveSolutionBoundTracker)
+            // See https://github.com/SonarSource/sonarlint-visualstudio/issues/3886
+            TriggerUpdate().Forget();
+        }
+
+        private void OnPreSolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs e)
+        {
+            TriggerUpdate().Forget();
         }
 
-        public async Task TriggerUpdate()
+        private void OnPreSolutionBindingUpdated(object sender, EventArgs e) => TriggerUpdate().Forget();
+
+        private async Task TriggerUpdate()
```

Comment:
One question though, Do we want the file to only get written once or do we want it to get updated whenever the user does an update or binds? probably only once right? if so I'll add the dispose call back into this method.

Reply:
Yep, only once sounds right. We can always change it in a future release if necessary.